engine/uState.ml

```diff
@@ -236,22 +237,23 @@ let process_universe_constraints ctx cstrs =
       else 
           match cst with
           | ULe (l, r) ->
-            if UGraph.check_leq univs l r then
-              (* Keep Prop/Set <= var around if var might be instantiated by prop or set
-                 later. *)
-              match Universe.level l, Universe.level r with
-              | Some l, Some r ->
-                Constraint.add (l, Le, r) local
-              | _ -> local
-            else
-              begin match Universe.level r with
-              | None -> user_err Pp.(str "Algebraic universe on the right")
-              | Some r' ->
-                if Level.is_small r' then
+            begin match Univ.Universe.level r with
+            | None ->
+              if UGraph.check_leq univs l r then local
+              else user_err Pp.(str "Algebraic universe on the right")
+            | Some r' ->
+              if Level.is_small r' then
                   if not (Universe.is_levels l)
                   then
+                    if UGraph.check_leq univs l r then local
+                    else
                     raise (UniverseInconsistency (Le, l, r, None))
```

Comment:
At the beginning of this chunk we have `is_small r'` and `not (is_levels l)`, so `l <= r` is impossible, so we may replace the `if check_leq l r then local else raise` by just `raise`